### PR TITLE
Add missing properties exportation to Texture.toJSON()

### PR DESCRIPTION
**Pull request by takahirox**
This PR adds missing properties exportation to `Texture.toJSON()` and updates  `ObjectLoader.parseTextures()` for it.

BTW, I'm thinking to export only non-default properties in `.toJSON()` like

    if ( this.name !== '' ) output.name = this.name;
    if ( this.mipmaps.length > 0 ) output.mipmaps = this.mipmaps.slice();
    if ( this.type !== UnsignedByteType ) output.type = this.type;

Pros: serialized data size can be smaller. Cons: information can be lost if we change the default values.

If you like, I'll make another PR.

### CODE REVIEW

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -564,13 +564,19 @@ Object.assign( ObjectLoader.prototype, {
 				}
 
 				if ( data.format !== undefined ) texture.format = data.format;
+				if ( data.type !== undefined ) texture.type = data.type;
+				if ( data.encoding ) texture.encoding = data.encoding;
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js]: Why not using this pattern?
```js
if ( data.encoding !== undefined ) texture.encoding = data.encoding;
```

File: src/loaders/ObjectLoader.js [hunk 2]
```diff
@@ -564,13 +564,19 @@ Object.assign( ObjectLoader.prototype, {
...
 
 				if ( data.minFilter !== undefined ) texture.minFilter = parseConstant( data.minFilter, TEXTURE_FILTER );
 				if ( data.magFilter !== undefined ) texture.magFilter = parseConstant( data.magFilter, TEXTURE_FILTER );
 				if ( data.anisotropy !== undefined ) texture.anisotropy = data.anisotropy;
+				if ( data.mipmaps !== undefined ) texture.mipmaps = data.mipmaps;
 
 				if ( data.flipY !== undefined ) texture.flipY = data.flipY;
 
+				if ( data.premultiplyAlpha ) texture.premultiplyAlpha = data.premultiplyAlpha;
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js]: ```if ( data.premultiplyAlpha !== undefined )``` ?

File: src/loaders/ObjectLoader.js [hunk 3]
```diff
@@ -564,13 +564,19 @@ Object.assign( ObjectLoader.prototype, {
 				}
 
 				if ( data.format !== undefined ) texture.format = data.format;
+				if ( data.type !== undefined ) texture.type = data.type;
+				if ( data.encoding ) texture.encoding = data.encoding;
 
 				if ( data.minFilter !== undefined ) texture.minFilter = parseConstant( data.minFilter, TEXTURE_FILTER );
 				if ( data.magFilter !== undefined ) texture.magFilter = parseConstant( data.magFilter, TEXTURE_FILTER );
 				if ( data.anisotropy !== undefined ) texture.anisotropy = data.anisotropy;
+				if ( data.mipmaps !== undefined ) texture.mipmaps = data.mipmaps;
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js]: AFAIK, these are manually created mipmaps. How can these data be represented in serialized JSON?

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (takahirox) [reply src/loaders/ObjectLoader.js]: Oops, I guess I was too sleepy when I was making this PR. Thanks.

File: src/loaders/ObjectLoader.js (hunk 3 above)
Comment (takahirox) [reply src/loaders/ObjectLoader.js]: Yup, it's for manual mipmaps generation. According to the source code, .mipmaps format is either one of 

    [ image1, image2, ... ]

    [ { width: width1, height: height1, data: typedArray1 },
      { width: width2, height: height2, data: typedArray2 } ]

Then, it should be like this in JSON?

    [ uuid1, uuid2 ]
    images = [
        { uuid: uuid1, ... },
        { uuid: uuid2, ... },
    ]

    [
      { width: width1, height: height1, data:  [0, 1, 2, ... ], type: typedArrayType1 },
      { width: width2, height: height2, data:  [0, 1, 2, ... ], type: typedArrayType2 },
    ]

Comment (takahirox) [reply src/loaders/ObjectLoader.js]: More precisely, for this style

    [ image1, image2, ... ]

image* should be ImageData / HTMLImageElement / HTMLCanvasElement / HTMLVideoElement / ImageBitmap.

For this style

    [ { width: width1, height: height1, data: typedArray1 },
      { width: width2, height: height2, data: typedArray2 } ]

typedArray* should be ArrayBufferView

https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D
https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLTextures.js

Any type of ImageData / HTMLImageElement / HTMLCanvasElement / HTMLVideoElement / ImageBitmap can be serialized by `getDataURL()` of `Texture.toJSON()`.

https://github.com/mrdoob/three.js/blob/dev/src/textures/Texture.js#L133

Comment (Mugen87) [reply src/loaders/ObjectLoader.js]: If @mrdoob thinks it worth, then let's give it a try.

File: src/loaders/ObjectLoader.js [hunk 4]
```diff
@@ -564,13 +564,55 @@ Object.assign( ObjectLoader.prototype, {
...
+						Uint8Array: Uint8Array,
+						// Workaround for IE11 pre KB2929437. See #11440
+						Uint8ClampedArray: typeof Uint8ClampedArray !== 'undefined' ? Uint8ClampedArray : Uint8Array,
+						Int16Array: Int16Array,
+						Uint16Array: Uint16Array,
+						Int32Array: Int32Array,
+						Uint32Array: Uint32Array,
+						Float32Array: Float32Array,
+						Float64Array: Float64Array
+					};
```
Comment (takahirox) [inline src/loaders/ObjectLoader.js]: I copied `TYPED_ARRAYS` from `BufferGeometryLoader`. Wondering if we can share this table across modules...

File: src/loaders/ObjectLoader.js (hunk 3 above)
Comment (takahirox) [reply src/loaders/ObjectLoader.js]: Implemented `.mipmaps` serialization so far.

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
@mrdoob I think this PR can be merged.

**Comment by Mugen87** [comment]
@mrdoob I think it's okay to merge this PR.

**Pull request merged**